samples/reactjs-todo/client/components/journey/form.js

```diff
@@ -132,12 +143,20 @@ export default function Form({ action, bottomMessage, followUp, topMessage }) {
         return <TextOutput callback={cb} key={`textOutput-${idx}`} />; //For TextOutput callbacks, 'input' field comes empty which leads to a unique-key-prop error
       case 'ConfirmationCallback':
         return <Confirmation callback={cb} inputName={name} key={name} />;
+      case 'SelectIdPCallback':
+        return <IdentityProvider callback={cb} inputName={name} key={name} />;
       default:
         // If current callback is not supported, render a warning message
         return <Unknown callback={cb} key={`unknown-${idx}`} />;
     }
   }
-
+  function isIdentityProviderLogin(callback) {
+    return callback
+      ?.getOutputByName('providers', [])
+      .filter((provider) => provider.provider !== 'localAuthentication').length > 0
+      ? true
+      : false;
+  }
```

Comment:
I'm a bit confused here. Is the intention to identify if this is social login by testing if it has 1 or more social identity providers, once you filter out `localAuthentication`? I ask because if there's a `SelectIdPCallback` in the step, this will always be true. The `SelectIdPCallback` will never have just `localAuthentication` in it. So, the existence of `SelectIdPCallback` is itself enough to detect social login.

Reply:
right, removed!